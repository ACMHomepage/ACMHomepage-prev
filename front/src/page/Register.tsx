import { useEffect, useState } from 'react';
import { Lock, User, Mail, Eye } from 'lucide-react';
import { useNavigate, Link } from 'react-router-dom';

import { useSelector } from '../hooks';

import { utilMainPart, boxSx } from '../config';
import { setBorder, setColor, setFlex } from '../util/theme';
import { selectState, AuthStateEnum, useSignUp } from '../store/authSlice';

// Import components.
import Header from '../component/Header';
import Button from '../component/Button';
import Input from '../component/Input';

import { URL as signInUrl } from './SignIn';

export const URL = '/register';

/**
 * Page `Sign`. Handle the sign in / sign on.
 */
export default () => {
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const authState = useSelector(selectState);
  const navigator = useNavigate();

  const signUp = useSignUp();

  useEffect(() => {
    if (authState === AuthStateEnum.LoggedWithInfo) {
      // Just go back.
      navigator(-1);
    }
  });

  // TODO: Return error if the name, email or password is empty.
  // TODO: We do not handler if email is already in database.
  // TODO: We do not show the helpful message in different state.
  return (
    <Header.Space sx={{ ...utilMainPart }}>
      <div sx={{ ...boxSx, ...setFlex({ direction: 'column', gap: '1rem' }) }}>
        <div
          sx={{
            textAlign: 'center',
            fontSize: '4xl',
            fontWeight: 'h3',
            mb: '1.5rem',
          }}
        >
          ACM Homepage
        </div>
        <Input
          placeholder="User name"
          startIcon={User}
          value={[name, (event) => setName(event.target.value)]}
        />
        <Input
          placeholder="Email"
          startIcon={Mail}
          value={[email, (event) => setEmail(event.target.value)]}
        />
        <Input
          placeholder="Password"
          startIcon={Lock}
          endIcon={Eye}
          value={[password, (event) => setPassword(event.target.value)]}
        />
        <Button
          sx={{
            height: '2.5rem',
            fontSize: 'lg',
            mt: '2rem',
            ...setColor({ bg: 'bg-0', color: 'fg-0', hover: { bg: 'bg-2' } }),
            ...setBorder({ width: '2px', color: 'bg-4' }),
            '&:hover': {
              ...setBorder({ width: '2px', color: 'bg-5' }),
            },
          }}
          onClick={() => signUp(name, email, password)}
        >
          Resgister
        </Button>
      </div>
      <div sx={boxSx}>
        Already have an account? Try to <Link to={signInUrl}>sign in</Link>.
      </div>
    </Header.Space>
  );
};
