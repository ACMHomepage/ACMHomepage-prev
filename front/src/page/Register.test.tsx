import Register from './Register'

test('After Register return to',() =>{
	expect(useEffect()).toEqual(-1);
});
