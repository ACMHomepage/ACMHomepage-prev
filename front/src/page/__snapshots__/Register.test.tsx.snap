// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make sure \`Resgister\` looks same as before. 1`] = `
<div>
  <div
    class="css-5m2jzx-HeaderSpace"
  >
    <div
      class="css-f0zbmj"
    >
      <div
        class="css-1ji8ini"
      >
        ACM Homepage
      </div>
      <label
        class="css-f5kwef"
      >
        <svg
          class="css-phcl6x-NickNameInput"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 21v-2a4 4 0 00-4-4H8a4 4 0 00-4 4v2"
          />
          <circle
            cx="12"
            cy="7"
            r="4"
          />
        </svg>
        <input
          class="css-1y8h48x-NickNameInput"
          placeholder="Nickname"
          value=""
        />
      </label>
      <label
        class="css-f5kwef"
      >
        <svg
          class="css-tzl219-EmailInput"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
          />
          <polyline
            points="22,6 12,13 2,6"
          />
        </svg>
        <input
          class="css-23obw9-EmailInput"
          placeholder="Email"
          value=""
        />
      </label>
      <label
        class="css-f5kwef"
      >
        <svg
          class="css-mahqe2-PasswordInput"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="11"
            rx="2"
            ry="2"
            width="18"
            x="3"
            y="11"
          />
          <path
            d="M7 11V7a5 5 0 0110 0v4"
          />
        </svg>
        <input
          class="css-ew39hz-PasswordInput"
          placeholder="Password"
          value=""
        />
        <svg
          class="css-1gd8vyt-PasswordInput"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
          />
          <circle
            cx="12"
            cy="12"
            r="3"
          />
        </svg>
      </label>
      <button
        class="css-yo3zwe"
      >
        Register
      </button>
    </div>
    <div
      class="css-10ij95f"
    >
      Already have an account? Try to 
      <a
        href="/signin"
      >
        sign in
      </a>
      .
    </div>
  </div>
</div>
`;
